src/peer_manager.rs

```diff
@@ -562,15 +564,15 @@ impl PeerManager {
     }
 
     /// Return the PeerIds of nodes bearing the names.
-    pub fn get_peer_ids(&self, names: &[XorName]) -> Vec<PeerId> {
+    pub fn get_peer_ids(&self, names: &HashSet<XorName>) -> Vec<PeerId> {
         names.iter()
             .filter_map(|name| self.peer_map.get_by_name(name).and_then(Peer::peer_id))
             .cloned()
             .collect()
     }
 
     /// Return the PublicIds of nodes bearing the names.
-    pub fn get_pub_ids(&self, names: &[XorName]) -> Vec<PublicId> {
+    pub fn get_pub_ids(&self, names: &HashSet<XorName>) -> HashSet<PublicId> {
```

Comment:
I think the `names.contains()` call isn't really natural: Shouldn't that just be checked in the `else` inside the iteration?

Anyway, I don't feel strongly about that one. Happy to leave it as is.